src/anm/animation/animation.js

```diff
@@ -166,11 +167,34 @@ Animation.prototype.traverse = function(visitor, data) {
  *
  * @param {Function} visitor
  * @param {anm.Element} visitor.child
+ * @param {Boolean} visitor.return if `false` returned, stops the iteration. no-`return` or empty `return` both considered `true`.
  * @param {Object} [data]
  */
 Animation.prototype.each = function(visitor, data) {
     for (var i = 0, tlen = this.tree.length; i < tlen; i++) {
-        visitor(this.tree[i], data);
+        if (visitor(this.tree[i], data) === false) break;
```

Comment:
Needless to say that typically visitors or any types of consumers return
true saying "it's done", like, "consumed"

On Mon, May 25, 2015, 8:53 PM Dmitry Skavish <notifications@github.com>
wrote:

> In src/anm/animation/animation.js
> <https://github.com/Animatron/player/pull/306#discussion_r30999144>:
>
> >   * @param {Object} [data]
> >   */
> >  Animation.prototype.each = function(visitor, data) {
> >      for (var i = 0, tlen = this.tree.length; i < tlen; i++) {
> > -        visitor(this.tree[i], data);
> > +        if (visitor(this.tree[i], data) === false) break;
>
> comparing to false is somewhat bad on many levels
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Animatron/player/pull/306/files#r30999144>.
>